components/server/src/main/java/com/hotels/styx/server/track/CurrentRequest.java

```diff
@@ -0,0 +1,50 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.server.track;
+
+import com.hotels.styx.api.HttpRequest;
+
+/**
+ * Bean that represent the current request.
+ */
+public class CurrentRequest {
+
+    private final String request;
+    private final long startingTimeMillies;
+    private Thread currentThread;
```

Comment:
I think the `currentThread` needs to be volatile, as it is set from the different threads, and read from another.